extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/Http.java

```diff
@@ -0,0 +1,44 @@
+package io.quarkus.vertx.http.runtime;
+
+import org.eclipse.microprofile.config.ConfigProvider;
+
+import io.quarkus.runtime.annotations.ConfigDocIgnore;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+import io.smallrye.config.ConfigMapping;
+import io.smallrye.config.SmallRyeConfig;
+import io.smallrye.config.WithDefault;
+
+@ConfigMapping(prefix = "quarkus.http")
+@ConfigRoot(phase = ConfigPhase.RUN_TIME)
+public interface Http {
+    /**
+     * The HTTP host
+     * <p>
+     * In dev/test mode this defaults to localhost, in prod mode this defaults to 0.0.0.0
+     * <p>
+     * Defaulting to 0.0.0.0 makes it easier to deploy Quarkus to container, however it
+     * is not suitable for dev/test mode as other people on the network can connect to your
+     * development machine.
+     */
+    @ConfigDocIgnore
+    String host();
+
+    /**
+     * The HTTP port
+     */
+    @ConfigDocIgnore
+    @WithDefault("8080")
+    int port();
```

Comment:
I'm a bit confused, what's the goal here? 

The configured port may not be the actual port (if set to 0 or -1 for example). Also what about the other servers?

Reply:
I meant - do not use this as a way to get the actual HTTP port and host.

Reply:
A lot of extensions rely on `VertxHttpConfiguration` to retrieve the host and the port, and we had some trouble in the past because we changed this class and broke these extensions. Configuration classes are not exactly part of the public API, so this was a first attempt to discuss having something that extensions can use safely.

Other servers can come later.